src/ImageProcessor.Web.Plugins.PostProcessor/PostProcessor.cs

```diff
@@ -41,116 +41,142 @@ internal static class PostProcessor
-            }
-
-            string sourceFile = null, destinationFile = null;
-            var timeout = PostProcessorBootstrapper.Instance.Timout;
-            try
-            {
-                // Save source file length
-                var length = stream.Length;
-
-                // Create a temporary source file with the correct extension
-                var tempSourceFile = Path.GetTempFileName();
-                sourceFile = Path.ChangeExtension(tempSourceFile, extension);
-                File.Move(tempSourceFile, sourceFile);
-
-                // Give our destination file a unique name
-                destinationFile = sourceFile.Replace(extension, "-out" + extension);
-
-                // Get processes to start
-                var processStartInfos = GetProcessStartInfos(extension, length, sourceFile, destinationFile).ToList();
-                if (processStartInfos.Count > 0)
+                string sourceFile = null, destinationFile = null;
+                try
                 {
-                    // Save the input stream to our source temp file for post processing
-                    using (var fileStream = File.Create(sourceFile))
-                    {
-                        await stream.CopyToAsync(fileStream).ConfigureAwait(false);
-                    }
+                    // Get temporary file names
+                    var tempPath = Path.GetTempPath();
+                    var tempFile = Path.GetFileNameWithoutExtension(Path.GetRandomFileName());
+                    sourceFile = Path.Combine(tempPath, Path.ChangeExtension(tempFile, extension));
+                    destinationFile = Path.Combine(tempPath, Path.ChangeExtension(tempFile + "-out", extension));
 
-                    // Create cancellation token with timeout
-                    using (var cancellationTokenSource = new CancellationTokenSource(timeout))
+                    // Get processes to start
+                    var processStartInfos = GetProcessStartInfos(extension, length, sourceFile, destinationFile).ToList();
+                    if (processStartInfos.Count > 0)
                     {
-                        foreach (var processStartInfo in processStartInfos)
+                        // Save the input stream to our source temp file for post processing
+                        var sourceFileInfo = new FileInfo(sourceFile);
+                        using (var fileStream = sourceFileInfo.Create())
                         {
-                            // Use destination file as new source (if previous process created one).
-                            if (File.Exists(destinationFile))
-                            {
-                                File.Copy(destinationFile, sourceFile, true);
-                            }
+                            // Try to keep the file in memory and ensure it's not indexed
+                            sourceFileInfo.Attributes |= FileAttributes.Temporary | FileAttributes.NotContentIndexed;
```

Comment:
No need to flush the data back to mass storage and to index it, see: https://docs.microsoft.com/en-us/dotnet/api/system.io.fileattributes.